### ISSUE TITLE
Events and contacts cannot be deleted using a Windows Phone 8.1 iCloud account

### ISSUE THREAD

**Issue by konmax** [Kozea/Radicale#263]
If the client is a Windows Phone 8.1 using the iCloud account as mentioned in the documentation from Radicale, events and contacts can be created and changed, but not deleted.

**Comment by konmax** [comment]
Hi,

I uploaded a corresponding log showing this issue here:

http://www.appdata.de/radicale.txt

Konrad

**Comment by liZe** [comment]
In the logs, the lines of the ics events sent by the client seem to have an extra empty line between each line. I thought that it was caused by the server hosted on a Windows machine, but I've tried and I can't reproduce this problem with Thunderbird+Radicale on my Windows virtual machine.

Could you please try to create an event with another client and see if the extra empty lines are in the logs too with your Radicale hosted on Windows?

**Comment by konmax** [comment]
No, this only seems to happen with Windows Phone clients. Android and Lightning works OK.

Konrad


Am 21. September 2015 18:11:42 MESZ, schrieb Guillaume Ayoub <notifications@github.com>:

**Comment by konmax** [comment]
One more thing: On the file system level (the collection file) I can see this empty lines only after contact photo streams in vcf files. The rest of the file and ics completely looks OK.

Konrad


Am 21. September 2015 20:50:16 MESZ, schrieb Konrad Hammerer <konrad.hammerer@appdata.de>:

**Comment by konmax** [comment]
Again more info:

Except the issue (events/contacts cannot be deleted) and the additional new line after a foto stream in an vcf file, everything works just fine with Windows Phone 8.1 as client and Radicale on Windows.

Konrad

**Comment by liZe** [comment]
So, I suppose that the new line after the photo is finally a separate issue. Could you create another issues with relevent logs, please?

If the extra empty lines are just in the logs and not in the file, the issue is probably caused by something else. I can't reproduce the error without the Windows phone client, even with Radicale on Windows, so I suppose that there's something else I haven't seen.

Could you please try to add some debug lines in the code, so that I can find what's going on? In `radicale/__init__.py`:
- line 383 (before `if item:`, same indentation), 3 new lines: `log.LOGGER.debug(item)`, `log.LOGGER.debug(environ["PATH_INFO"])`, `log.LOGGER.debug(collection.path)`
- line 387 (before `if if_match in ("*", item.etag):`, same indentation), 1 new line: `log.LOGGER.debug(if_match)`

You should then have these values in the logs when you try to delete an event.

Thank you for your patience, we'll get rid of this bug soon!

**Comment by konmax** [comment]
Hi,

Will do all that later today. 

One thought: Do you have an up and running Radicale server where I can connect with my phone. It might help you to debug this in a better and more efficient way.

Konrad




Am 22. September 2015 11:44:08 MESZ, schrieb Guillaume Ayoub <notifications@github.com>:

**Comment by konmax** [comment]
Hi,

Am 22.09.2015 um 11:44 schrieb Guillaume Ayoub:
>
> So, I suppose that the new line after the photo is finally a separate 
> issue. Could you create another issues with relevent logs, please?
>

Done:
https://github.com/Kozea/Radicale/issues/315


> If the extra empty lines are just in the logs and not in the file, the 
> issue is probably caused by something else. I can't reproduce the 
> error without the Windows phone client, even with Radicale on Windows, 
> so I suppose that there's something else I haven't seen.
>
> Could you please try to add some debug lines in the code, so that I 
> can find what's going on? In |radicale/__init__.py|:
>
>   * line 383 (before |if item:|, same indentation), 3 new lines:
>     |log.LOGGER.debug(item)|,
>     |log.LOGGER.debug(environ["PATH_INFO"])|,
>     |log.LOGGER.debug(collection.path)|
>   * line 387 (before |if if_match in ("*", item.etag):|, same
>     indentation), 1 new line: |log.LOGGER.debug(if_match)|
>

My code seems to be an older version and looks slightly different:

With these changes Radicale does not start anymore:

         else:
             # Try to get an item matching the path
             item = collection.get_item(
                 xmlutils.name_from_path(environ["PATH_INFO"], collection))

*log.LOGGER.debug(item)**
**        log.LOGGER.debug(environ["PATH_INFO"])**
**        log.LOGGER.debug(collection.path)*
         if item:
             # Evolution bug workaround
             etag = environ.get("HTTP_IF_MATCH", 
item.etag).replace("\\", "")
*log.LOGGER.debug(etag)*
             if etag == item.etag:

(I have etag instead of if_match like you mentioned)

Please advice!

Thanks
Konrad

**Comment by konmax** [comment]
Better formatted (previous post was sent through mail):

...
        else:
            # Try to get an item matching the path
            item = collection.get_item(
                xmlutils.name_from_path(environ["PATH_INFO"], collection))

        log.LOGGER.debug(item)
        log.LOGGER.debug(environ["PATH_INFO"])
        log.LOGGER.debug(collection.path)
        if item:
            # Evolution bug workaround
            etag = environ.get("HTTP_IF_MATCH", item.etag).replace("\\", "")
            log.LOGGER.debug(etag)
            if etag == item.etag:
...

**Comment by liZe** [comment]
Thank you for your new bug report!

Your idea is good, I've launched a server at http://kozea.fr:5232/, you can try to create and delete an event, I'll get the logs!

**Comment by konmax** [comment]
OK. Which collection (URL)/user/password can I use?

Konrad

Am 23.09.2015 um 11:39 schrieb Guillaume Ayoub:

**Comment by liZe** [comment]
It's public, there's no user/password. You can try `/test/test.ics/`.

**Comment by konmax** [comment]
Hi,

Tried it with the browser (http://kozea.fr:5232/test/test.ics/) -> works.
Tried it with the phone -> does not work. Not sure why, there is no 
message. Maybe the phone does not accept public collections. Can you 
create one with a user and password? Or do you see anything on the 
server why it does not work?

Konrad

Am 23.09.2015 um 15:36 schrieb Guillaume Ayoub:

**Comment by konmax** [comment]
Hi,

Maybe Windows Phone has a problem if there is no user and password?! At least I cannot create the account without credentials. Did you see anything in the logs why I was not able to connect. I was able to connect successfully using a browser and Thunderbird, though. 

Thx
Konrad



Am 23. September 2015 15:36:05 MESZ, schrieb Guillaume Ayoub <notifications@github.com>:

**Comment by liZe** [comment]
The server was down, it's up again, you can try with the phone, I'll check the logs and tell you what I get!

**Comment by konmax** [comment]
Hi,

The server was working for the browser and Thunderbird at the time I 
tried to connect.

However, I cannot do another test before Saturday. I'm in France for a 
business trip today and tomorrow with no access to the Windows Phone.

Keep the server running. Will connect on Saturday and get back to you...

Thx
Konrad


Am 24.09.2015 um 10:35 schrieb Guillaume Ayoub:

**Comment by konmax** [comment]
Hi!

I tried to connect an hour ago but it still does not work.

In order to make progress with this issue, I uploaded my __init__.py file to my server:

http://www.appdata.de/init.zip

Please download it and make the necessary changes to get all the debug messages you need. I will then add it to my installation, run the test with the phone and send you the logs.

Thx
Konrad

**Comment by konmax** [comment]
Seems to be related to #117 and is already fixed with 1.0

Thanks
Konrad
